### ISSUE TITLE
Delegate Collection::Set to ::Set using method missing

### ISSUE THREAD

**Issue by sbellware** [eventide-project/collection#4]
Any errors resulting from no method on ::Set should be re-raised as emanating from Collection::Set

**Comment by smdern** [comment]
Would delegating to ::Set effectively remove the need to implement Enumerable, equality, clear, etc ?

**Comment by sbellware** [comment]
I had totally forgotten that it was already an enumerable. I'm not sure `clear` is part of `Enumerable`. If not, `clear` might be the only thing that needs to be implemented explicitly. The equality operator probably needs an explicit implementation due to the need to consider the type parameter.

**Comment by smdern** [comment]
:+1: Should we audit the methods that are overriding `Enumerable` implementations? Things like `count` should probably use `Enumerable#count` since it also supports taking a block. Currently it's aliased to `length`.

**Comment by sbellware** [comment]
@smdern Yes, absolutely. It was built on the quick and now deserves more real attention. Did you want to do it?

**Comment by smdern** [comment]
@sbellware sure, I don't mind doing it.
